### PR TITLE
Implemented: barcode identifier selector and scanning based on selected identifier (#845)

### PR DESCRIPTION
**Pull request by amansinghbais**
### Related Issues

#845

### Short Description and Why It's Useful

Added support for the barcode identification scanning.

### Screenshots of Visual Changes before/after (If There Are Any)

Screenshot from 2024-11-06 12-33-20

### Contribution and Currently Important Rules Acceptance

- [x] I read and followed [contribution rules](https://github.com/hotwax/fulfillment#contribution-guideline)

### CODE REVIEW

File: src/views/Settings.vue [hunk 1]
```diff
@@ -286,7 +289,8 @@ export default defineComponent({
       orderLimitType: 'unlimited',
       fulfillmentOrderLimit: "" as number | string,
       facilityGroupDetails: {} as any,
-      isEComInvEnabled: false
+      isEComInvEnabled: false,
+      barcodeContentMessage: translate("Only allow received quantity to be incremented by scanning the barcode of products. If the identifier is not found, the scan will default to using the internal name.", { space: '<br /><br />' })
```
Comment (ravilodhi) [inline src/views/Settings.vue]: "Only allow received quantity......"  should be changed to "Only allow shipped quantity"

File: src/store/modules/util/actions.ts [hunk 2]
```diff
@@ -585,9 +585,134 @@ const actions: ActionTree<UtilState, RootState> = {
...
+      } else {
+        dispatch("cretaBarcodeIdentificationPref");
+      }
+    } catch(err) {
+      console.error(err)
+      commit(types.UTIL_BARCODE_IDENTIFICATION_PREF_UPDATED, "internalName")
+    }
+  },
+
+  async cretaBarcodeIdentificationPref({ commit }) {
```
Comment (ravilodhi) [inline src/store/modules/util/actions.ts]: Correct typo.

File: src/store/modules/util/UtilState.ts [hunk 3]
```diff
@@ -13,4 +13,5 @@ export default interface UtilState {
   facilities: any;
   shipmentGatewayConfigs: any;
   isForceScanEnabled: boolean;
+  barcodeIdentificationPref: any;
```
Comment (ymaheshwari1) [inline src/store/modules/util/UtilState.ts]:
```suggestion
  barcodeIdentificationPref: string;
```

File: src/store/modules/util/actions.ts [hunk 4]
```diff
@@ -585,9 +585,134 @@ const actions: ActionTree<UtilState, RootState> = {
...
+        "settingValue": "internalName"
+      }  
+
+      await UtilService.createBarcodeIdentificationPref(params) as any
+    } catch(err) {
+      console.error(err)
+    }
+
+    // not checking for resp success and fail case as every time we need to update the state with the
+    // default value when creating a scan setting
```
Comment (ymaheshwari1) [inline src/store/modules/util/actions.ts]:
```suggestion
    // default value when creating a store setting
```

File: src/store/modules/util/actions.ts [hunk 5]
```diff
@@ -585,9 +585,134 @@ const actions: ActionTree<UtilState, RootState> = {
...
+      console.error(err)
+    }
+    commit(types.UTIL_BARCODE_IDENTIFICATION_PREF_UPDATED, prefValue)
+  },
   
   async updateForceScanStatus({ commit }, payload) { 
     commit(types.UTIL_FORCE_SCAN_STATUS_UPDATED, payload)
+  },
+
+  async updateBarcodeIdentificationPref({ commit }, payload) {
```
Comment (ymaheshwari1) [inline src/store/modules/util/actions.ts:714]: This action is not used anywhere, we should use this method wherever we are updating the state for barcode identifier, also we need to clear the barcode identifier value of logout.

File: src/store/modules/util/index.ts [hunk 6]
```diff
@@ -21,7 +21,8 @@ const utilModule: Module<UtilState, RootState> = {
     productStores: [],
     facilities: [],
     shipmentGatewayConfigs: [],
-    isForceScanEnabled: false
+    isForceScanEnabled: false,
+    barcodeIdentificationPref: ""
```
Comment (ymaheshwari1) [inline src/store/modules/util/index.ts]: As we have defined `internalName` as default value for barcodeIdentifier we can have the same value here in state as its default declaration

**Review by ymaheshwari1 — CHANGES_REQUESTED** [review]

**Comment by amansinghbais** [comment]
Closing this PR since latest commit got stuck in github glitch, Processing updates

Created new PR for this: https://github.com/hotwax/fulfillment/pull/851

**Pull request closed without merging**